frontend/src/hooks/queryBuilder/useAutoComplete.ts

```diff
@@ -0,0 +1,174 @@
+import { QUERY_BUILDER_OPERATORS_BY_TYPES } from 'constants/queryBuilder';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { checkStringEndWIthSpace } from '../../utils/checkStringEndWIthSpace';
+import { useFetchKeysAndValues } from './useFetchKeysAndValues';
+import { useSetCurrentKeyAndOperator } from './useSetCurrentKeyAndOperator';
+import { useTag } from './useTag';
+import { useTagValidation } from './useTagValidation';
+
+type Option = {
+	value: string;
+	selected?: boolean;
+};
+
+type ReturnT = {
+	handleSearch: (value: string) => void;
+	handleClearTag: (value: string) => void;
+	handleSelect: (value: string) => void;
+	handleKeyDown: (e: React.KeyboardEvent) => void;
+	options: Option[];
+	tags: string[];
+	searchValue: string;
+	isFilter: boolean;
+};
+
+export type KeyType = {
+	key: string;
+	dataType: 'string' | 'boolean' | 'number';
+	type: string;
+};
+
+export const useAutoComplete = (): ReturnT => {
+	const [searchValue, setSearchValue] = useState('');
```

Comment:
can you check if we can use antd form for this one

Reply:
We can but I think this is not better idea on this level. We also have general context with values state and there we can use it but I can see why. But if you want, of course I can rewrite it on ant form